### PR TITLE
PM-15593 Standard Login UI is shown when item has no Auth key added

### PR DESCRIPTION
**Pull request by dan-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/jira/software/c/projects/BW/boards/93?selectedIssue=PM-15593
## 📔 Objective

Only display totp ciphers when the field is a totp field and there is a totp code

- Added getFilteredCiphersForTotpField method
- Implemented getFilteredCiphersForTotpField on update list method
- Updated tests to not always have a top field

## 📸 Screenshots
Screenshot 2024-12-16 at 12 39 12 PM
Screenshot 2024-12-16 at 12 39 25 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/browser/src/autofill/overlay/inline-menu/pages/list/__snapshots__/autofill-inline-menu-list.spec.ts.snap [hunk 1]
```diff
@@ -681,352 +681,35 @@ exports[`AutofillInlineMenuList initAutofillInlineMenuList the list of ciphers f
...
-      >
-        <button
-          aria-label="fillCredentialsFor "
-          class="fill-cipher-button inline-menu-list-action"
-          tabindex="-1"
-        >
-          <span
-            aria-hidden="true"
-            class="cipher-icon bwi bw-icon"
+            style="background-image: url(https://jest-testing-website.com/image.png);"
```
Comment (coltonhurst) [inline apps/browser/src/autofill/overlay/inline-menu/pages/list/__snapshots__/autofill-inline-menu-list.spec.ts.snap:692]: Up to @jprusik but do we want to rely on an external website here + in the other areas? (I know this is the `spec.ts.snap` file, there may be a convo I wasn't aware of)

**Comment by dan-livefront** [comment]
BIT Run: https://github.com/bitwarden/browser-interactions-testing/actions/runs/12378521222/job/34550582564

File: apps/browser/src/autofill/overlay/inline-menu/pages/list/__snapshots__/autofill-inline-menu-list.spec.ts.snap (hunk 1 above)
Comment (jprusik) [reply apps/browser/src/autofill/overlay/inline-menu/pages/list/__snapshots__/autofill-inline-menu-list.spec.ts.snap:692]: We use this domain in our current cipher mocks for testing, but I wouldn't mind moving to something under our ownership. Might be a good feature extension to [test-the-web](https://github.com/bitwarden/test-the-web)... 🤔 

In any case, this is fine for now

Comment (jprusik) [reply apps/browser/src/autofill/overlay/inline-menu/pages/list/__snapshots__/autofill-inline-menu-list.spec.ts.snap:692]: Tech debt task created for this:
https://bitwarden.atlassian.net/browse/PM-16204

File: apps/browser/src/autofill/overlay/inline-menu/pages/list/autofill-inline-menu-list.ts [hunk 2]
```diff
@@ -412,6 +412,26 @@ export class AutofillInlineMenuList extends AutofillInlineMenuPageElement {
...
+  /**
+   * Filters the ciphers to include only TOTP-related ones if the field is a TOTP field.
+   *
+   * @param ciphers - The list of ciphers to filter.
+   * @returns The filtered list of ciphers or the original list if no filtering is needed.
+   */
+  private getFilteredCiphersForTotpField(ciphers: InlineMenuCipherData[]): InlineMenuCipherData[] {
+    if (!ciphers?.length) {
+      return ciphers || [];
+    }
```
Comment (jprusik) [inline apps/browser/src/autofill/overlay/inline-menu/pages/list/autofill-inline-menu-list.ts:424]: were we not doing this check upstream, or is this purely defensive?

File: apps/browser/src/autofill/overlay/inline-menu/pages/list/__snapshots__/autofill-inline-menu-list.spec.ts.snap (hunk 1 above)
Comment (coltonhurst) [reply apps/browser/src/autofill/overlay/inline-menu/pages/list/__snapshots__/autofill-inline-menu-list.spec.ts.snap:692]: Thanks @jprusik!

File: apps/browser/src/autofill/overlay/inline-menu/pages/list/autofill-inline-menu-list.ts (hunk 2 above)
Comment (dan-livefront) [reply apps/browser/src/autofill/overlay/inline-menu/pages/list/autofill-inline-menu-list.ts]: This was purely defensive.

**Review by jprusik — APPROVED** [review]

**Review by differsthecat — APPROVED** [review]
Approving after merges from main; the only changes since @jprusik 's approval were merge conflict fixes in unit tests.

**Pull request merged**